Source/WTF/wtf/text/WTFString.h

```diff
@@ -512,6 +512,17 @@ inline void String::clearImplIfNotShared()
         m_impl = nullptr;
 }
 
+inline String String::substring(unsigned position, unsigned length) const
+{
+    if (!m_impl)
+        return { };
+
+    if (!position && length >= this->length())
```

Comment:
I suggest `m_impl->length()` instead of `this->length()` for consistency with the rest of the function and to **guarantee** we don't get an extra null check (I am sure it was optimized out).